ui/store/actions.js

```diff
@@ -1116,6 +1126,39 @@ export function updateMetamaskState(newState) {
       type: actionConstants.UPDATE_METAMASK_STATE,
       value: newState,
     });
+
+    // Check that the transaction was not submitted successfully, and remove it from failed transactions if it was.
+    const transactionIdsToRemove = Object.keys(
+      transactionsToDisplayOnFailure,
+    ).filter((id) => {
+      const foundTx = currentNetworkTxList
+        .map((tx, index) => {
+          return { id: tx.id, status: tx.status, index };
+        })
+        .find((currentTx) => {
+          return id === currentTx.id;
+        });
+
+      // check if the status of foundTx has changed
+      if (foundTx) {
+        const newTx = newNetworkTxList[foundTx.index];
```

Comment:
There is no guarantee that the transaction at this index is the correct transaction. We need to find this transaction by id.